pkg/prometheus/promcfg.go

```diff
@@ -2376,9 +2376,16 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 			configs[i] = []yaml.MapItem{
 				{
 					Key:   "role",
-					Value: strings.ToLower(config.Role),
+					Value: strings.ToLower(string(config.Role)),
 				},
 			}
+
+			if len(config.Selectors) != 0 {
+				configs[i] = append(configs[i], yaml.MapItem{
+					Key:   "selectors",
+					Value: config.Selectors,
+				})
+			}
```

Comment:
(nit) I'd be explicit and not rely on the fact that the serialized CRD struct is identical to the Prometheus struct.
```suggestion
			selectors := make([]yaml.MapItem, len(config.Selectors))
			for i, s := range config.Selectors {
				selectors[i] = []yaml.MapItem{
					{
						Key:   "role",
						Value: strings.ToLower(string(s.Role)),
					},
										{
						Key:   "label",
						Value: s.Label,
					},
										{
						Key:   "field",
						Value: s.Field,
					},
				}
			}
			if len(selectors) > 0 {
				configs[i] = append(configs[i], yaml.MapItem{
					Key:   "selectors",
					Value: selectors,
				})
			}
```